docs/react-v9/contributing/testing-with-jest.md

```diff
@@ -0,0 +1,234 @@
+  - [Mock functions](https://jestjs.io/docs/mock-function-api)
+  - [The `jest` object](https://jestjs.io/docs/jest-object) including fake timer APIs ([more info](https://jestjs.io/docs/timer-mocks))
+
+Note that you do not need to import the assertions or the Jest APIs; they should be available automatically through the included typings.
+
+A basic test example:
+
+```ts
+describe('thing', () => {
+  it('does something', () => {
+    expect(thing()).toEqual(aValue);
+  });
+});
+```
+
+## Testing components
+
+In cases where you need to automate a component and validate it performs correctly, you can use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) to mount components, evaluate DOM structure, and simulate events.
+
+Here's a very basic example of what a test might look like (we'll go over this in more detail later):
+
+```tsx
+const onClick = jest.fn();
+const { getByRole } = render(<Button onClick={onClick}>This is a button</Button>);
+
+userEvent.click(getByRole('button'));
+expect(onClick).toHaveBeenCalled();
+```
+
+### Philosophy
+
+The [guide on migrating from Enzyme](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/) nicely sums up Testing Library's philosophy, and how it differs from the way tests tended to be written with Enzyme:
+
+> The primary purpose of React Testing Library is to increase confidence in your tests by **testing your components in the way a user would use them.** Users don't care what happens behind the scenes, they just see and interact with the output. Instead of accessing the components' internal APIs or evaluating their state, you'll get more confidence by writing your tests based on the component output.
+
+Some examples of this philosophy:
+
+- Prefer [finding elements like a user would](https://testing-library.com/docs/queries/about/#priority) (by role, label, or text) rather than relying on `className`, `id`, or DOM structure. This approach helps encourage writing UI that's accessible to all users.
+- Prefer interact with elements the way the user would, by clicking or using the keyboard, rather than using imperative APIs.
+  - [`@testing-library/user-event`](https://testing-library.com/docs/ecosystem-user-event/) more realistically simulates the full sequence of events that occur during user interaction.
+
+Testing Library's APIs also make it _easier_ to write realistic, user-focused tests compared to Enzyme.
+
+### Recommended reading
+
+This page will only go into basics of using Testing Library (and some [common pitfalls](#tips-and-faq)).
+
+For a more detailed run-through of how to write tests, check out this [tutorial](https://www.robinwieruch.de/react-testing-library/) recommended in Testing Library's docs. (One difference: we typically use the queries returned by `render()`, rather than the global queries from `screen` like the tutorial uses.)
+
+It's highly recommended that you check out Testing Library's docs as well!
+
+- Basics
```

Comment:
I'm not sure if it;s good idea to copy paste outlines of existing docs, can we simplify this to one link ?

Reply:
I'm not sure what you mean here? Generally speaking, I like having a basic overview in a single doc that can help an intern understand something, and then link out to the big thick heavy manual.

Reply:
All of this is already outlined and well documented within RTL docs 

<img width="391" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/98969208-1d52-4d96-9c27-4efca2f0f350">

having manually maintained copy of well documented docs for 3rd party tool is not worth the effort of keeping it up to date, one link (https://testing-library.com/docs/react-testing-library/intro) with some encouragement to explore all the chapters is everything one should need.

Reply:
Okay - I've removed the bulk of that and referred to the Jest and RTL docs.